File: 백준/Silver/20546. 🐜 기적의 매매법 🐜/🐜 기적의 매매법 🐜.cc
#include <iostream>
#include <vector>
using namespace std;


int main() {
    
    int n;
    cin >> n;
    vector <int> v(14);
    for (auto& i : v)
    {
        cin >> i;
    }

    int j_money = n, s_money = n, j_stock = 0, s_stock = 0;
    for (int i = 0; i < 14; i++)
    {
        while (j_money >= v[i])
        {
            j_stock++;
            j_money -= v[i];
        }

        if (2 < i)
        {
            if (v[i - 3] < v[i - 2] && v[i - 2] < v[i - 1]) //3일 연속 상승
            {
                while (s_stock > 0)
                {
                    s_stock--;
                    s_money += v[i];
                }
            }
            else if (v[i - 3] > v[i - 2] && v[i - 2] > v[i - 1]) //3일 연속 하락
            {
                while (s_money >= v[i])
                {
                    s_stock++;
                    s_money -= v[i];
                }
            }
        }
    }
    int j_result = j_money + j_stock * v[13];
    int s_result = s_money + s_stock * v[13];

    if (j_result > s_result)
        cout << "BNP";
    else if (j_result < s_result)
        cout << "TIMING";
    else
        cout << "SAMESAME";
    return 0;
}

File: 백준/Silver/20546. 🐜 기적의 매매법 🐜/🐜 기적의 매매법 🐜.js
const fs = require('fs')
const filepath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = fs.readFileSync(filepath).toString().split('\n');
const seedMoney = +input[0];
const arr = input[1].split(' ').map(Number);

let jMoney = seedMoney;
let sMoney = seedMoney;
let jStock = 0;
let sStock = 0;


function buyStock(money, price, stock)
{
    while(money >= price){
        stock++;
        money -= price;
    }
    return [money, stock]
}

function sellStock(money, price, stock)
{
    while(stock > 0){
        stock--;
        money += price;
    }
    return [money, stock]
}

for(let i = 0; i < 14; i++){
    [jMoney, jStock] = buyStock(jMoney, arr[i], jStock);
    if(i > 2){
        if(arr[i - 3] < arr[i - 2] && arr[i - 2] < arr[i - 1]) //3일 연속 증가
            [sMoney, sStock] = sellStock(sMoney, arr[i], sStock);

        if(arr[i - 3] > arr[i - 2] && arr[i - 2] > arr[i - 1]) //3일 연속 하락
            [sMoney, sStock] = buyStock(sMoney, arr[i], sStock);
    }
}
const jResult = jMoney + jStock * arr[13]; 
const sResult = sMoney + sStock * arr[13];

if (jResult > sResult)
    console.log("BNP");
else if (jResult < sResult)
    console.log("TIMING");
else
    console.log("SAMESAME");